src/DSManager.cpp

```diff
@@ -1296,7 +1586,7 @@ void DSManager::loadPublisher(
     {
         // try load from profile
         if (xmlparser::XMLP_ret::XML_OK !=
-            xmlparser::XMLProfileManager::fillPublisherAttributes(std::string(profile_name), *pubatts))
+                xmlparser::XMLProfileManager::fillPublisherAttributes(std::string(profile_name), *pubatts))
```

Comment:
About this methods maybe you can use `DomainParticipant::get_publisher_qos_from_profile` that internally calls this method and translates the Attributes to QoS. You also have available `DomainParticipant::geet_subscriber_qos_from_profile` and `DomainParticipant::get_topic_qos_from_profile`